### PR TITLE
implements process eventhandler

### PR DESCRIPTION
**Pull request by matheus-marcelinux**
Changes related to process eventhandler.

### CODE REVIEW

**Comment by sauljabin** [comment]
Hi @matheus-marcelinux thanks for sharing,

- The changes have to be part of the template folder, you committed over the example plugin (I think I will remove it :thinking: )
- Also, can you please explain to us a little more about how this change will work and how we can test it?

again thanks a lot.

**Comment by matheus-marcelinux** [comment]
Hi @sauljabin. 

I've made the changes in template/plugin.symbolic.name and after runned ./plugin-scaffold seems to generate correct the plugin, maybe i'm missing something ? 

The changes are related to the process event handler, using UUID or ClassName. When registering an event using one of process event topics (BEFORE_PROCESS, AFTER_PROCESS, POST_PROCESS), the core method registerProcessEvent is called.
When a process event is dispatched and execHandler is invoked, the method doHandleEvent is called passing a ProcessInfo object, which may be retrivied by calling getProcessInfo() method in CustomEvent. 

To register a process event handler in EventFactory.java:

`registerEvent(IEventsTopics.BEFORE_PROCESS, "UUID or ClassName", MyProcessEventHandler.class);`

In MyProcessEventHandler.java:

```
@Override
protected void doHandleEvent() {
	ProcessInfo pi = getProcessInfo();
        log.info("Process Class: " + pi.getClassName());
}
```

Ps: i've changed all tableName refereces to filter, just to be according to osgi term.

**Comment by sauljabin** [comment]
Thank you so much @matheus-marcelinux I checked the files and I see it at template folder, thanks for the explanation, We will (@ocurieles) test and then merge your changes

File: template/plugin.symbolic.name/src/plugin.root/base/CustomEventFactory.java [hunk 1]
```diff
@@ -78,19 +93,22 @@ private void execEventHandler(Event event, EventHandlerWrapper eventHandlerWrapp
 	 * @param tableName    Table name
```
Comment (sauljabin) [inline template/plugin.symbolic.name/src/plugin.root/base/CustomEventFactory.java:93]: @matheus-marcelinux same parameter name in the doc

File: idempiere-plugin-template/com.ingeint.template/src/com/ingeint/template/base/CustomEventFactory.java [hunk 2]
```diff
@@ -78,19 +93,22 @@ private void execEventHandler(Event event, EventHandlerWrapper eventHandlerWrapp
 	 * @param tableName    Table name
```
Comment (sauljabin) [inline idempiere-plugin-template/com.ingeint.template/src/com/ingeint/template/base/CustomEventFactory.java:93]: @matheus-marcelinux same parameter name in the doc

File: idempiere-plugin-template/com.ingeint.template/src/com/ingeint/template/base/CustomEventFactory.java [hunk 3]
```diff
@@ -48,27 +49,41 @@ protected void doHandleEvent(Event event) {
...
 		if (eventType.equals(eventHandlerWrapper.getEventTopic())) {
-			if (eventHandlerWrapper.getTableName() != null) {
+			if (eventHandlerWrapper.getFilter() != null) {
+				if (isProcessHandler(eventType)) {
+					ProcessInfo pi = getProcessInfo(event);
+					if(pi.getAD_Process_UU().equals(eventHandlerWrapper.getFilter())
+					|| pi.getClassName().equals(eventHandlerWrapper.getFilter())) {
+						execEventHandler(event, eventHandlerWrapper, pi);
+					}
+				} else {
```
Comment (sauljabin) [inline idempiere-plugin-template/com.ingeint.template/src/com/ingeint/template/base/CustomEventFactory.java:59]: @matheus-marcelinux indentation

**Comment by sauljabin** [comment]
@matheus-marcelinux thank you so much for contributing

**Pull request merged**